server/src/main/java/org/elasticsearch/search/fetch/PreloadedFieldLookupProvider.java

```diff
@@ -26,14 +27,15 @@
  */
 class PreloadedFieldLookupProvider implements LeafFieldLookupProvider {
 
-    Map<String, List<Object>> storedFields;
-    LeafFieldLookupProvider backUpLoader;
-    Supplier<LeafFieldLookupProvider> loaderSupplier;
+    private Set<String> preloadedStoredFields;
+    private Map<String, List<Object>> storedFields;
+    private LeafFieldLookupProvider backUpLoader;
+    private Supplier<LeafFieldLookupProvider> loaderSupplier;
```

Comment:
I took the chance to make these private and add package private setter/getter methods when needed. I find that it clarifies who does what and when.

Reply:
:heart:

Reply:
Just a question...do I understand it correctly that `preloadedStoredFieldValues.keySet()` is the same as `preloadedStoredFieldNames`? Or is one a subset of the other?

Because later on I see we do

```
if (preloadedStoredFieldNames.get().contains(field)) {
            fieldLookup.setValues(preloadedStoredFieldValues.get(field));
```

which looks like as "if the name is there and the field was preloaded then just get the preloaded values"...

Reply:
if they were the same we would not need  a separate set. preloadedStoredFieldNames includes all the fields that we know we will attempt to load for all documents. Those include fields that don't have a value, while preloadedStoredFieldValues contains only those fields that were found in the current doc. 

The overhead was caused by trying to load `_ignored` and `_routing` ad-hoc when requested for all docs that did not have a value for them.

Reply:
Ok thank you.